### PR TITLE
Fix NODE_BINARY_URL extraction

### PR DESCRIPTION
**Pull request by djmarcin**
Fixes https://github.com/heroku/heroku-buildpack-nodejs/issues/927

Maintains backward compatibility with anyone who set up a custom tarball with an embedded misnamed folder like `node-v-linux-x64` by stripping the initial path component unconditionally rather than baking in an assumption about what that path component will look like.

### CODE REVIEW

**Comment by djmarcin** [comment]
@dzuelke Sorry to ping directly, but just wondering if this was missed. Can you help with getting this PR approved and merged? Is there any CLA I need to sign to commit to the repo? Thanks!

File: lib/binaries.sh [hunk 1]
```diff
@@ -115,9 +115,8 @@ install_nodejs() {
   if [ "$code" != "200" ]; then
     echo "Unable to download node: $code" && false
   fi
-  tar xzf /tmp/node.tar.gz -C /tmp
   rm -rf "${dir:?}"/*
-  mv /tmp/node-v"$number"-"$os"-"$cpu"/* "$dir"
```
Comment (joshwlewis) [inline lib/binaries.sh:120]: It looks like you are dropping the only usage of `$os` and `$cpu` here. Mind dropping the assignments a few lines up (`os=$(get_os)` and `cpu=$(get_cpu)` )?

**Review by joshwlewis — CHANGES_REQUESTED** [review]
This looks great, thanks. Probably a performance improvement, too. 

Would you mind adding a `CHANGELOG.md` entry and rebasing, so we can get the checks all green?

File: lib/binaries.sh (hunk 1 above)
Comment (djmarcin) [reply lib/binaries.sh:120]: Done

**Comment by djmarcin** [comment]
> This looks great, thanks. Probably a performance improvement, too.
> 
> Would you mind adding a `CHANGELOG.md` entry and rebasing, so we can get the checks all green?

Added the changelog entry and rebased. Thanks for taking a look!

**Comment by djmarcin** [comment]
CircleCI is failing with
```
Traceback (most recent call last):
	3: from /usr/local/bundle/gems/heroku_hatchet-7.1.3/etc/ci_setup.rb:16:in `<main>'
	2: from /usr/local/bundle/gems/heroku_hatchet-7.1.3/etc/ci_setup.rb:16:in `open'
	1: from /usr/local/bundle/gems/heroku_hatchet-7.1.3/etc/ci_setup.rb:19:in `block in <main>'
/usr/local/bundle/gems/heroku_hatchet-7.1.3/etc/ci_setup.rb:19:in `fetch': key not found: "HEROKU_API_USER" (KeyError)
bundler: failed to load command: hatchet (/usr/local/bundle/bin/hatchet)
```

Is this what is meant by `First-time contributors need a maintainer to approve running workflows.`?

**Pull request merged**